From 975042cc72ddca334f1e351dce6b76194146ed99 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 4 May 2018 11:49:35 -0700
Subject: [PATCH 1/2] mips: Disable attribute-alias warnings

This warning is seen with gcc-8
include/linux/syscalls.h:211:18: error: 'sys_cachectl' alias between functions of incom
patible types 'long int(char *, int,  int)' and 'long int(long int,  long int,  long int)'

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 arch/mips/kernel/Makefile | 2 ++
 arch/mips/mm/Makefile     | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/mips/kernel/Makefile b/arch/mips/kernel/Makefile
index f10e1e15e1c6..eb92e52eb3db 100644
--- a/arch/mips/kernel/Makefile
+++ b/arch/mips/kernel/Makefile
@@ -2,6 +2,8 @@
 #
 # Makefile for the Linux/MIPS kernel.
 #
+CFLAGS_signal.o		+= $(call cc-disable-warning, attribute-alias)
+CFLAGS_syscall.o	+= $(call cc-disable-warning, attribute-alias)
 
 extra-y		:= head.o vmlinux.lds
 
diff --git a/arch/mips/mm/Makefile b/arch/mips/mm/Makefile
index c463bdad45c7..b7f9ef80dac7 100644
--- a/arch/mips/mm/Makefile
+++ b/arch/mips/mm/Makefile
@@ -2,6 +2,7 @@
 #
 # Makefile for the Linux/MIPS-specific parts of the memory manager.
 #
+CFLAGS_cache.o			+= $(call cc-disable-warning, attribute-alias)
 
 obj-y				+= cache.o dma-default.o extable.o fault.o \
 				   gup.o init.o mmap.o page.o page-funcs.o \
-- 
2.5.0

